core/message-handling.go

```diff
@@ -157,13 +157,16 @@ func defaultIncomingMessageHandler(id uint32, log messagelog.MessageLog, config
 	}
 	clientStates := clientstate.NewProvider(clientStateOpts...)
 	peerStates := peerstate.NewProvider()
+	viewState := viewstate.New()
 
 	captureSeq := makeRequestSeqCapturer(clientStates)
 	prepareSeq := makeRequestSeqPreparer(clientStates)
 	retireSeq := makeRequestSeqRetirer(clientStates)
 	startReqTimer := makeRequestTimerStarter(clientStates, handleReqTimeout, logger)
 	stopReqTimer := makeRequestTimerStopper(clientStates)
 	captureUI := makeUICapturer(peerStates)
+	provideView := viewState.WaitAndHoldActiveView
+	waitView := viewState.WaitAndHoldView
```

Comment:
It sounds to me surprising to see `WaitAndHoldActiveView` is used to get current view. But I can't clearly say what is better, but `WaitAndHoldActiveView` could be named better?

Reply:
Here, `provideView` just allows to get the current active view. But whenever the current active view is required, it can only be acquired when it becomes available, i.e. in normal mode only. Once the view number is acquired, it must be guaranteed not to change, thus has to be released afterwards. The name `WaitAndHoldActiveView` attempts to reflect this logic.

The name `viewProvider` is more concise and generic, since the code using it (`makeRequestApplier`) doesn't really care about those details - it only has to obey the requirement to release any acquired view. In contrast, the code supplying this abstraction (`defaultIncomingMessageHandler`) has to ensure that this primitive is used consistently with the other things, thus it has to deal with `viewstate` package and consider all those details.

We could also introduce trivial functions `makeViewProvider`, `makeViewWaiter`, that will take `viewState` as a parameter. But since `WaitAndHoldActiveView`/`WaitAndHoldView` matches `viewProvider`/`viewWaiter` in definition, it can be simply used directly.